src/TestFramework/Config/TestFrameworkConfigLocator.php

```diff
@@ -24,20 +26,33 @@ public function __construct(string $configDir)
     public function locate(string $testFrameworkName, string $customDir = null): string
     {
         $dir = $customDir ?: $this->configDir;
+        $triedFiles = [];
 
-        foreach (['xml', 'yml'] as $extension) {
+        foreach ($this->getDefaultExtensions() as $extension) {
             $conf = sprintf('%s/%s.%s', $dir, $testFrameworkName, $extension);
-            $altConf = sprintf('%s/%s.dist.%s', $dir, $testFrameworkName, $extension);
 
             if (file_exists($conf)) {
                 return realpath($conf);
-            } elseif (file_exists($conf . '.dist')) {
-                return realpath($conf . '.dist');
-            } elseif (file_exists($altConf)) {
-                return realpath($altConf);
             }
+
+            $triedFiles[] = sprintf('%s.%s', $testFrameworkName, $extension);
         }
 
-        throw new \RuntimeException(sprintf('Unable to locate %s(.dist).(xml|yml)(.dist) file.', $testFrameworkName));
+        throw LocatorException::multipleFilesDoNoExist(
+            $dir,
+            $triedFiles
+        );
+    }
+
+    private function getDefaultExtensions() : array
```

Comment:
Can't this be a `const DEFAULT_EXTENSIONS = [ ... ];`? [Like it's done elsewhere.](https://github.com/infection/infection/blob/85ca245272a70db7ecd8e6db0de73fd3b3ec5a19/src/Mutator/Util/MutatorProfile.php#L16)

Reply:
+1 to extract to the constant

Reply:
It could be, but I'm thinking future proofing and maintaining solid coding. If the frameworks every overrode this class, the method is more effective. For example, since codeception uses .dist.yml, but not .yml.dist, the codeception testing code override this class and say specifically .yml.dist only. Const is too tightly coupled in my experience.

Reply:
If you want to override class like this you'll need to create your own Infection library as you don't have a chance to replace it via your own.

Also I'm pretty sure that we will drop 7.0 support at the end of this year and with 7.1 we can use constant visibility.

Reply:
I'm actually referring specifically to the in-project frameworks (`./src/TestFramework`)

Reply:
> For example, since codeception uses .dist.yml, but not .yml.dist, the codeception testing code override this class and say specifically .yml.dist only. Const is too tightly coupled in my experience.

1. private method can't be overriden
2. it's possible to override const
3. since all such arrays we store in constants, let's store this one as well to be consistent across the project